docs/important/relevant_functionalities_de.md

```diff
@@ -0,0 +1,1250 @@
+
+1. Identifier von dem Konsumenten identifizieren 
+2. Löschen der Daten und des Service Accounts des Konsumenten
+
+Ein Beispiel für das `service-account-remove`-Skript könnte folgendermaßen aussehen:
+
+`remove-sa.sh`
+
+```bash
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#1) Identifier von dem Konsumenten identifizieren
+SERVICE="${1}"
+if [ X"${SERVICE}" = X"" ]; then
+  echo "usage remove-sa.sh servicename"
+  exit 1
+fi
+
+#2) Löschen des Service Accounts des Verbrauchers
+# ... Implementiere die Dogu-spezifische Logik hier
+```
+
+In Schritt 1 wird der Name des Konsumenten-Dogus ausgelesen (z. B. "redmine"), i. d. R. um den Service Account des Dogus zu identifizieren.
+
+In Schritt 2 wird der Account aus der Datenbasis des Produzenten inkl. Zugangsdaten gelöscht.
+
+### Service Accounts konsumieren
+
+Es ist sehr einfach, bei einem Produzenten-Dogu einen Service Account anzufragen, da die Hauptarbeit beim Client und beim Produzenten liegt. Ein Konsumenten-Dogu muss lediglich in seiner `dogu.json` das gewünschte Produzenten-Dogu als [ServiceAccount]() nennen:
+
+```json
+{
+  ...,
+  "ServiceAccounts": [
+    {
+      "Type": "produzentendogu",
+      "Kind": "dogu"
+    }
+  ],
+  ...
+}
+```
+
+Bei einem Service Account handelt es sich um eine besondere Form der Abhängigkeit. Trotzdem ist es sinnvoll, das Produzenten-Dogu in seiner [Dependency-Liste]() zu führen. So wird in den unterschiedlichen Phasen einer Dogu-Installation sichergestellt, dass das Produzenten-Dogu wirklich zur Benutzung bereitsteht:
+
+
+```json
+{
+  ...,
+  "Dependencies": [
+    {
+      "name": "produzentendogu",
+      "type": "dogu"
+    }
+  ],
+  "ServiceAccounts": [
+    {
+      "Type": "produzentendogu",
+      "Kind": "dogu"
+    }
+  ],
+  ...
+}
+```
+
+## Dogu-Upgrades
+
+Das Upgrade eines Dogus kann mit `cesapp upgrade dogu <doguname>` durchgeführt werden. Dabei verwaltet die cesapp alle wichtigen Schritte, um das Dogu erfolgreich zu upgraden.
+
+Der Upgrade-Prozess enthält u.A. folgende Schritte:
+- Anzeigen von Upgrade Notifications, falls vorhanden
+- Download des neuen Dogu-Images
+- Sicherstellen, dass das Dogu healthy ist
+- Sicherstellen, dass alle abhängigen Dogus verfügbar sind
+- Ausführen des Pre-Upgrade-Skripts im alten Dogu, falls vorhanden
+- Ersetzen des alten Dogus durch das neue
+- Starten des neuen Dogus
+- Ausführen des Post-Upgrade-Skripts im neuen Dogu, falls vorhanden
+
+Falls es beim Dogu-Upgrade notwendig ist, vor- oder nachbereitende Schritte durchzuführen, können optionale Pre- und Post-Upgrade-Skripte erstellt werden.
+Als Beispiel kann man eine Software betrachten, die in einer neueren Version eine andere Struktur der Datenbank benötigt.
+In einem solchen Fall ist es möglich, Upgrade-Skripte zu schreiben, um eine Migration der Datenbank nach einem Dogu-Upgrade zu starten.
+
+Außerdem kann man über wichtige Änderungen beim Upgrade eines Dogus durch ein Upgrade-Notification-Skript hinweisen lassen.
```

Comment:
I suggest to use either "Außerdem kann man sich ... hinweisen lassen" or "Außerdem kann man ... hinweisen"

Reply:
Als Entwickler willst du ja durch das Skript nicht DICH auf etwas hinweisen lassen, sondern die Anwender/Administratoren.